core/loadpoint.go

```diff
@@ -1409,8 +1409,7 @@ func (lp *LoadPoint) Update(sitePower float64, cheap bool, batteryBuffered bool)
 	mode := lp.GetMode()
 	lp.publish("mode", mode)
 
-	// read and publish meters first
-	lp.UpdateChargePower()
+	// read and publish meters first- charge power has already been updated by the site
 	lp.updateChargeCurrents()
```

Comment:
I would even go one step further and 
- also remove lp.updateChargeCurrents here 
- rename this function from "Update" to "ReevaluateAndPublishState()"   (best name I could think of ;))

-> LoadPoint doesn't update input values here anymore, this function is turned into a 'pure' processing function.

- site.Update() gets the lp.updateChargeCurrents() together with UpdateChargePower(). This 'twin functions' thus will be called together at the same place again.

Then the order + responsibility would be even more clean:
- Site triggers all updates for ChargePower+ChargeCurrents at the very beginning
- loadpoint 'just' processes and publishes state based on filled values.